lib/ansible/modules/database/postgresql/postgresql_query.py

```diff
@@ -0,0 +1,344 @@
+    named_args:
+      a_value: "positional string value 1"
+      b_value: "positional string value 2"
+
+# Run queries from a '.sql' file and assign result in a fact available at
+# for the rest of the ansible runtime.
+- postgresql_query:
+    db: acme
+    user: django
+    password: ceec4eif7ya
+    query: SELECT * FROM some_table WHERE a_column=%(a_value)s AND b_column=%(b_value)s
+    named_args:
+      a_value: "positional string value 1"
+      b_value: "positional string value 2"
+    fact: my_key
+'''
+
+RETURN = '''
+query_results:
+    description: List of dictionaries in column:value form representing returned rows.
+    returned: changed
+    type: list
+    sample: [{"Column": "Value1"},{"Column": "Value2"}]
+row_count:
+    description: number of affected rows by query, if applicable
+    returned: changed
+    type: int
+    sample: 5
+'''
+
+HAS_PSYCOPG2 = False
+try:
+    import psycopg2
+    import psycopg2.extras
+except ImportError:
+    pass
+else:
+    HAS_PSYCOPG2 = True
+
+import traceback
+from ansible.module_utils.six import iteritems
+from ansible.module_utils.basic import AnsibleModule, get_exception, BOOLEANS
+import ansible.module_utils.postgres as pgutils
+
+
+def main():
+    argument_spec = pgutils.postgres_common_argument_spec()
+
+    argument_spec.update(dict(
+        db=dict(default=None),
+        query=dict(type="str"),
+        autocommit=dict(type='bool', default=False, choices=BOOLEANS),
+        positional_args=dict(type="list"),
+        named_args=dict(type="dict"),
+        fact=dict(default=None),
+    ))
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[
+            ["positional_args", "named_args"]
+        ],
+        supports_check_mode=True,
+    )
+
+    if not HAS_PSYCOPG2:
+        module.fail_json(msg="the python psycopg2 module is required")
+
+    changed = False
+
+    # To use defaults values, keyword arguments must be absent, so
+    # check which values are empty and don't include in the **kw
+    # dictionary
+    params_map = {
+        "login_host": "host",
+        "login_user": "user",
+        "login_password": "password",
+        "port": "port",
+        "db": "database",
+        "ssl_mode": "sslmode",
+        "ssl_rootcert": "sslrootcert"
+    }
+
+    kw = dict((params_map[k], v) for (k, v) in iteritems(module.params)
+              if k in params_map and v != '' and v is not None)
+
+    # If a login_unix_socket is specified, incorporate it here.
+    is_localhost = "host" not in kw or kw["host"] == "" or kw["host"] == "localhost"
+
+    if is_localhost and module.params["login_unix_socket"] != "":
+        kw["host"] = module.params["login_unix_socket"]
+
+    try:
+        pgutils.ensure_libs(sslrootcert=module.params.get('ssl_rootcert'))
+        db_connection = psycopg2.connect(**kw)
+
+        # Enable autocommit on demand only so we can create databases
+        if module.params['autocommit']:
```

Comment:
I agree, but i can't find where this would be configured on the server side.
Maybe [here](https://www.postgresql.org/docs/9.5/static/runtime-config-connection.html#RUNTIME-CONFIG-CONNECTION-SETTINGS) or [here](https://www.postgresql.org/docs/9.5/static/runtime-config-client.html)

We could always do something like:

``` python      
        if module.params['autocommit']:
            if psycopg2.__version__ >= '2.4.2':
                db_connection.autocommit = True
            else:
                db_connection.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
        else:
            if psycopg2.__version__ >= '2.4.2':
                db_connection.autocommit = False
```